### PR TITLE
consensus/pbft: set the header time and add some delay

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: consensus/pbft/backends/simple/backend.go [hunk 1]
```diff
@@ -130,6 +131,13 @@ func (sb *simpleBackend) Commit(proposal *pbft.Proposal) error {
 		sb.logger.Error("Failed to commit proposal since RequestContext cannot cast to *types.Block")
 		return errCastingRequest
 	}
+
+	// TODO: Find a better way to handle, not just wait here
+	delay := time.Unix(block.Header().Time.Int64(), 0).Sub(time.Now())
+	select {
+	case <-time.After(delay):
```
Comment (tailingchen) [inline consensus/pbft/backends/simple/backend.go]: ` <-time.After(delay)` is enough.

Comment (markya0616) [inline consensus/pbft/backends/simple/backend.go]: OK

File: consensus/pbft/backends/simple/engine.go [hunk 2]
```diff
@@ -286,26 +290,32 @@ func (sb *simpleBackend) newChannels() {
...
+	parent := chain.GetHeader(header.ParentHash, number-1)
+	if parent == nil {
+		return nil, consensus.ErrUnknownAncestor
+	}
+	block, err := sb.updateBlock(parent, block)
+	if err != nil {
+		return nil, err
+	}
+
+	// wait for our time
```
Comment (yutelin) [inline consensus/pbft/backends/simple/engine.go]: make a clearer comment

File: consensus/pbft/backends/simple/engine.go [hunk 3]
```diff
@@ -286,26 +290,32 @@ func (sb *simpleBackend) newChannels() {
 // Seal generates a new block for the given input block with the local miner's
 // seal place on top.
 func (sb *simpleBackend) Seal(chain consensus.ChainReader, block *types.Block, stop <-chan struct{}) (*types.Block, error) {
+	header := block.Header()
```
Comment (yutelin) [inline consensus/pbft/backends/simple/engine.go]: so the following code is to prepare a block for proposal? add comment

File: consensus/pbft/backends/simple/engine.go [hunk 4]
```diff
@@ -330,6 +340,31 @@ func (sb *simpleBackend) Seal(chain consensus.ChainReader, block *types.Block, s
...
+	if len(block.Transactions()) == 0 {
+		period = sb.config.BlockPauseTime
+	} else {
+		period = sb.config.BlockPeriod
+	}
+
+	// set header timestamp
+	header := block.Header()
+	header.Time = new(big.Int).Add(parent.Time, new(big.Int).SetUint64(period))
+	if header.Time.Int64() < now().Unix() {
```
Comment (yutelin) [inline consensus/pbft/backends/simple/engine.go]: call now() once and use the same `now` cross this func. every time you call now() is a different timestamp.

File: consensus/pbft/backends/simple/engine.go (hunk 2 above)
Comment (markya0616) [inline consensus/pbft/backends/simple/engine.go]: OK

File: consensus/pbft/backends/simple/engine.go (hunk 4 above)
Comment (markya0616) [inline consensus/pbft/backends/simple/engine.go]: OK

**Pull request merged**